backend/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSender.java

```diff
@@ -0,0 +1,153 @@
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@ApplicationScoped
+public class EmailSender {
+
+    private final Logger logger = Logger.getLogger(this.getClass().getName());
+
+    static final String BOP_APITOKEN_HEADER = "x-rh-apitoken";
+    static final String BOP_CLIENT_ID_HEADER = "x-rh-clientid";
+    static final String BOP_ENV_HEADER = "x-rh-insights-env";
+    static final String BODY_TYPE_HTML = "html";
+    static final ZoneOffset UTC = ZoneOffset.UTC;
+
+    @Inject
+    @BopWebClient
+    WebClient bopWebClient;
+
+    @ConfigProperty(name = "processor.email.bop_url")
+    String bopUrl;
+
+    @ConfigProperty(name = "processor.email.bop_apitoken")
+    String bopApiToken;
+
+    @ConfigProperty(name = "processor.email.bop_client_id")
+    String bopClientId;
+
+    @ConfigProperty(name = "processor.email.bop_env")
+    String bopEnv;
+
+    @Inject
+    WebhookTypeProcessor webhookSender;
+
+    @Inject
+    EndpointResources endpointResources;
+
+    @Inject
+    MeterRegistry registry;
+
+    private Counter processedCount;
+    private Timer processTime;
+
+    @PostConstruct
+    public void init() {
+        processedCount = registry.counter("processor.email.processed");
+        processTime = registry.timer("processor.email.process-time");
+    }
+
+    public Uni<NotificationHistory> sendEmail(User user, Action action, TemplateInstance subject, TemplateInstance body) {
+        final HttpRequest<Buffer> bopRequest = this.buildBOPHttpRequest();
+        LocalDateTime start = LocalDateTime.now(UTC);
+
+        return endpointResources.getOrCreateEmailSubscriptionEndpoint(action.getAccountId(), new EmailSubscriptionProperties())
+                .onItem().transformToUni(endpoint -> {
+                    Notification notification = new Notification(action, endpoint);
+
+                    // TODO Add recipients processing from policies-notifications processing (failed recipients)
+                    //      by checking the NotificationHistory's details section (if missing payload - fix in WebhookTypeProcessor)
+
+                    // TODO If the call fails - we should probably rollback Kafka topic (if BOP is down for example)
+                    //      also add metrics for these failures
+
+                    return webhookSender.doHttpRequest(
+                            notification,
+                            bopRequest,
+                            getPayload(user, action, subject, body)
+                    ).onItem().invoke(unused -> {
+                        processedCount.increment();
+                        processTime.record(Duration.between(start, LocalDateTime.now(UTC)));
```

Comment:
> It would be great if we could monitor with metrics the average execution time between the moment we want to send an email and the moment when the BOP request is actually completed.

@gwenneg 
Is this the right way of doing it?

Reply:
Yes that looks good. We don't really need that to run with `UTC` but I don't think using that ZoneOffset comes at a cost so you can leave it that way.